src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforcePushTopicListener.java

```diff
@@ -129,20 +129,95 @@ protected void customize(Request exchange) {
 
     // Now set up the Bayeux client itself
     BayeuxClient client = new BayeuxClient(oAuthInfo.getInstanceURL() + DEFAULT_PUSH_ENDPOINT, transport);
-    client.handshake();
-
     return client;
   }
 
-  private void waitForHandshake(BayeuxClient client,
-                                long timeoutInMilliseconds, long intervalInMilliseconds) {
+  public void createSalesforceListener() {
+    try {
+      bayeuxClient = getClient(credentials);
+    } catch (Exception exception) {
+      LOG.error("Exception while creating bayeux client ", exception);
+    }
+    bayeuxClient.getChannel(Channel.META_HANDSHAKE).addListener
+      ((ClientSessionChannel.MessageListener) (channel, message) -> {
+
+        boolean success = message.isSuccessful();
+        if (!success) {
+          String error = (String) message.get("error");
+          if (error != null) {
+            LOG.error("Error during HANDSHAKE: {}", error);
+          }
+
+          Exception exception = (Exception) message.get("exception");
+          if (exception != null) {
+            LOG.error("Exception during HANDSHAKE", exception);
+          }
+        }
+      });
+
+    bayeuxClient.getChannel(Channel.META_CONNECT).addListener(
+      (ClientSessionChannel.MessageListener) (channel, message) -> {
+
+        boolean success = message.isSuccessful();
+        if (!success) {
+          String error = (String) message.get("error");
+          Map<String, Object> advice = message.getAdvice();
+
+          if (error != null) {
+            LOG.error("Error during CONNECT: {}", error);
+            LOG.debug("Advice during CONNECT: {}", advice);
+          }
+          if (advice.get("reconnect").equals("handshake")) {
+            LOG.debug("Reconnecting to Salesforce Push Topic");
+            reconnectToTopic();
+          }
```

Comment:
why aren't we reconnecting when there are other types of errors?

Reply:
We specifically receive the following error message after the expiration of the 'sfdc-stream' cookie managed by the Salesforce load balancer; it has a 3hr expiration time, with the advice to reconnect using a handshake
[Mulesoft Help Center](https://help.mulesoft.com/s/article/Couldn-t-connect-to-Salesforce-streaming-server-because-a-403-error-has-been-returned-error-403-Unknown-client-Warning-messages-in-log)
```
{
  "error": "403::Unknown client",
  "successful": false,
  "advice": {
    "interval": 0,
    "reconnect": "handshake"
  }
}
```

The other types of error messages do not contain this reconnect advice.
A list of errors can be found on the Page No. 48 in this [document](https://blog.bessereau.eu/assets/pdfs/api_streaming.pdf)